operations/helm/charts/mimir-distributed/templates/memcached/_memcached-svc.tpl

```diff
@@ -17,7 +17,10 @@ metadata:
     {{- toYaml .service.annotations | nindent 4 }}
   namespace: {{ $.ctx.Release.Namespace | quote }}
 spec:
-  type: ClusterIP
+  type: {{ .service.type }}
+  {{- if semverCompare ">= 1.22-0" (include "mimir.kubeVersion" .) }}
+  internalTrafficPolicy: {{ .service.internalTrafficPolicy }}
+  {{- end }}
   clusterIP: None
```

Comment:
I wonder if we need to update services that are explicitly `headless` (note the `ClusterIP: None`).

Reply:
True, it doesn't make sense there.

https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/2086-service-internal-traffic-policy/README.md
`The internalTrafficPolicy field will not apply for headless Services or Services of type ExternalName.`